### PR TITLE
feat: colors convert from hex

### LINKED ISSUE #1492: Color32 from hex value
**Is your feature request related to a problem? Please describe.**

Most of the time when implementing a UI design, colors comes in hex values. Having to manually convert hex to RGBA is kind tedious and time consuming.

**Describe the solution you'd like**

new method in Color32

```rust
Color32::from_hex("#0d1117")
```

### PR DESCRIPTION
**Pull request by kkharji**
Closes #1492

### CODE REVIEW

**Comment by tami5** [comment]
fix: resulting colors are incorrect

File: epaint/src/color.rs [hunk 1]
```diff
@@ -74,6 +74,19 @@ impl Color32 {
...
+    pub fn from_hex(hex: &str) -> Result<Self, ParseIntError> {
+        let [r, g, b, _] = u32::from_str_radix(&hex.replace("#", ""), 16)?.to_be_bytes();
+        Ok(Self([r, g, b, 255]))
+    }
+
+    #[inline(always)]
+    // TODO: This is current identical to from_hex
+    pub fn from_unmultiplied(hex: &str) -> Result<Self, ParseIntError> {
+        let [r, g, b, a] = u32::from_str_radix(&hex.replace("#", ""), 16)?.to_be_bytes();
+        Ok(Self([r, g, b, a]))
```
Comment (emilk) [inline epaint/src/color.rs:87]:
```suggestion
        Ok(Self::from_rgba_unmultiplied(r, g, b, a))
```

File: epaint/src/color.rs [hunk 2]
```diff
@@ -74,6 +74,19 @@ impl Color32 {
         Self([r, g, b, 255])
     }
 
+    #[inline(always)]
+    pub fn from_hex(hex: &str) -> Result<Self, ParseIntError> {
+        let [r, g, b, _] = u32::from_str_radix(&hex.replace("#", ""), 16)?.to_be_bytes();
+        Ok(Self([r, g, b, 255]))
```
Comment (emilk) [inline epaint/src/color.rs:80]: I'd like for these functions to only accept `#rrggbb` and `#rrggbbaa`, and NOT e.g. `#123`. Also they need tests.

Comment (tami5) [reply epaint/src/color.rs:80]: like ensure that the hex is valid

```rust
pub fn validate_hex_string(hex: &str) -> Result<(), ()> {
    match &hex.len() {
        x if *x < 7 => return Err(()), // Truncated
        x if *x > 7 => return Err(()), // Overflow
        _ => {}
    };
    if &hex[0..1] != "#" {
        return Err(()); // Invalid
    }
    Ok(())
}
```

**Comment by tami5** [comment]
@emilk  🤔 that's odd one test fail with 
```rust
thread 'color::test_to_hex_to_rgba' panicked at 'assertion failed: `(left == right)`
  left: `Color32([20, 1, 7, 2])`,
 right: `Color32([245, 40, 145, 255])`', epaint/src/color.rs:587:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I explicitly set rgba to `Color32::from_rgba_unmultiplied(245, 40, 145, 2);` but it produced `[20, 1, 7, 2]`
 

btw 2 because I wanted  `0.2` not sure if this is correct

File: epaint/src/color.rs [hunk 3]
```diff
@@ -504,6 +518,75 @@ fn fast_round(r: f32) -> u8 {
...
+    let rgb = Color32::RED;
+    let hex = Color32::from_hex("#ff0000").unwrap();
+    assert_eq!(rgb, hex);
+}
+
+#[test]
+fn test_from_hex_to_rgba() {
+    let rgba = Color32::from_rgba_unmultiplied(245, 40, 145, 2);
+    let hex = Color32::from_hex("#f5289133").unwrap();
+    assert_eq!(rgba, hex);
```
Comment (emilk) [inline epaint/src/color.rs:587]:
```suggestion
    let rgba = Color32::from_rgba_unmultiplied(0xf5, 0x28, 0x91, 0x33);
    let hex = Color32::from_hex("#f5289133").unwrap();
    assert_eq!(rgba, hex);
```

**Comment by emilk** [comment]
The benefit with a macro (e.g. `egui::color!(#f5289133)`) is that errors can be caught at compilation time. If users want to have dynamic string they should probably be using one of the many css color parsing crates.

**Comment by emilk** [comment]
Closed by https://github.com/emilk/egui/pull/1596

**Pull request closed without merging**